onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -692,6 +725,95 @@ Status TensorProtoToTensor(const Env& env, const ORTCHAR_T* model_path,
   return Status::OK();
 }
 
+Status TensorProtoToTensor(const Env& /*env*/, const std::unordered_map<std::string, const void*>& external_data_map,
+                           const ONNX_NAMESPACE::TensorProto& tensor_proto,
+                           Tensor& tensor)
```

Comment:
There is a lot of code duplication in this function which will needlessly increase binary size. Can we re-use the code, perhaps adding default args?